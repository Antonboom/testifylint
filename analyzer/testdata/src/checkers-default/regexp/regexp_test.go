// Code generated by testifylint/internal/testgen. DO NOT EDIT.

package regexp

import (
	"regexp"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestRegexpChecker(t *testing.T) {
	var out string

	// Invalid.
	{
		assert.Regexp(t, regexp.MustCompile(`\[.*\] DEBUG \(.*TestNew.*\): message`), out)                                   // want "regexp: remove unnecessary regexp\\.MustCompile"
		assert.Regexpf(t, regexp.MustCompile(`\[.*\] DEBUG \(.*TestNew.*\): message`), out, "msg with args %d %s", 42, "42") // want "regexp: remove unnecessary regexp\\.MustCompile"
		assert.NotRegexp(t, regexp.MustCompile(`\[.*\] TRACE message`), out)                                                 // want "regexp: remove unnecessary regexp\\.MustCompile"
		assert.NotRegexpf(t, regexp.MustCompile(`\[.*\] TRACE message`), out, "msg with args %d %s", 42, "42")               // want "regexp: remove unnecessary regexp\\.MustCompile"
	}

	// Valid.
	{
		assert.Regexp(t, `\[.*\] DEBUG \(.*TestNew.*\): message`, out)
		assert.Regexpf(t, `\[.*\] DEBUG \(.*TestNew.*\): message`, out, "msg with args %d %s", 42, "42")
		assert.NotRegexp(t, `\[.*\] TRACE message`, out)
		assert.NotRegexpf(t, `\[.*\] TRACE message`, out, "msg with args %d %s", 42, "42")
	}
}
