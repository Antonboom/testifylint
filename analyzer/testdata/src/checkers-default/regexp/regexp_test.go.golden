// Code generated by testifylint/internal/testgen. DO NOT EDIT.

package regexp

import (
	"regexp"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestRegexpChecker(t *testing.T) {
	var out string

	// Invalid.
	{
		assert.Regexp(t, `\[.*\] DEBUG \(.*TestNew.*\): message`, out)                                   // want "regexp: remove unnecessary regexp\\.MustCompile"
		assert.Regexpf(t, `\[.*\] DEBUG \(.*TestNew.*\): message`, out, "msg with args %d %s", 42, "42") // want "regexp: remove unnecessary regexp\\.MustCompile"
		assert.NotRegexp(t, `\[.*\] TRACE message`, out)                                                 // want "regexp: remove unnecessary regexp\\.MustCompile"
		assert.NotRegexpf(t, `\[.*\] TRACE message`, out, "msg with args %d %s", 42, "42")               // want "regexp: remove unnecessary regexp\\.MustCompile"
	}

	// Valid.
	{
		assert.Regexp(t, `\[.*\] DEBUG \(.*TestNew.*\): message`, out)
		assert.Regexpf(t, `\[.*\] DEBUG \(.*TestNew.*\): message`, out, "msg with args %d %s", 42, "42")
		assert.NotRegexp(t, `\[.*\] TRACE message`, out)
		assert.NotRegexpf(t, `\[.*\] TRACE message`, out, "msg with args %d %s", 42, "42")
	}
}
